### PR TITLE
Add PR templates for deployments / deployment preparations.

### PR DESCRIPTION
**Pull request by jubeira**
# Description

We are generating deployment tasks and deploying contracts pretty often, and we are not always being consistent in the deployment steps (e.g. sometimes we deploy in the same PR where we generate the build info, sometimes we push action IDs afterwards, sometimes we go back and forth with the documentation, sometimes we even modify contract code, etc). While this is not a huge problem in most cases, it can be inconvenient e.g. when we need to go back and check the exact build info for a given contract, or when we need to branch a contract at a very specific point to address a very specific issue.

The proposal is to use standard checklists and make an explicit division between deployment preparation and deployment. Hopefully this will help reduce the back and forth, and also help document deployments.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [x] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- N/A Complex code has been commented, including external interfaces
- N/A Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

**Comment by jubeira** [comment]
Input and feedback is welcome :)

File: .github/PULL_REQUEST_TEMPLATE/development-template.md [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 ## Checklist:
 
-- [ ] The diff is legible and has no extraneous changes
+- [ ] The diff is legible and has no extraneous changes <!-- There shouldn't be any changes to contract code nor scripts in this PR -->
```
Comment (EndymionJkb) [inline .github/PULL_REQUEST_TEMPLATE/development-template.md:18]:
```suggestion
- [ ] The diff is legible and has no extraneous changes <!-- There shouldn't be any changes to contract code or scripts in this PR -->
```

File: .github/PULL_REQUEST_TEMPLATE/deployment-template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+- [ ] Contract(s) deployed to all target networks / testnets
+- [ ] TX IDs are properly generated
+- [ ] Action IDs are generated after deployment (if applicable) <!-- This includes contracts deployed by contracts (e.g. mock pools deployed by factories) -->
+- [ ] Contracts are verified in every network <!-- This includes contracts deployed by contracts (e.g. mock pools deployed by factories) -->
+
+## Code checklist:
+
+- [ ] The diff is legible and has no extraneous changes
+- [ ] The base branch is either `master`, or there's a description of how to merge
```
Comment (EndymionJkb) [inline .github/PULL_REQUEST_TEMPLATE/deployment-template.md:13]: Would this ever not be master? Maybe we don't need a code checklist section for this one, as there shouldn't be any code changes at this point.

**Review by EndymionJkb — COMMENTED** [review]
From the docs it looks like development-template.md still needs to be called pull_request_template.md, for it to be the default.

It says to add "?template=" to the URL request. How will that happen? Is the "Create Pull Request" going to have more options?

File: .github/PULL_REQUEST_TEMPLATE/deployment-template.md (hunk 2 above)
Comment (jubeira) [reply .github/PULL_REQUEST_TEMPLATE/deployment-template.md:13]: I agree; I thought about keeping the template just to enforce those two things explicitly, and force the submitter to double check.

If you think it's _too_ redundant, maybe we can just remove the checklist.

**Comment by jubeira** [comment]
> It says to add "?template=" to the URL request. How will that happen? Is the "Create Pull Request" going to have more options?

If I recall correctly, when you open a PR using the UI there should be an intermediate screen with all the template options, and you have to choose one of them. I don't think there's a 'default' one when there's more than one to choose from.

File: .github/PULL_REQUEST_TEMPLATE/deployment-template.md (hunk 2 above)
Comment (jubeira) [reply .github/PULL_REQUEST_TEMPLATE/deployment-template.md]: Done in https://github.com/balancer-labs/balancer-v2-monorepo/pull/2168/commits/3121c9b4b7f773aba3a37c387802bbbac845bb0a.

File: .github/PULL_REQUEST_TEMPLATE/deployment-preparation-template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+## Deprecating old tasks checklist: <!-- Only if applicable -->
+
+- [ ] Old task was moved to `deprecated/` directory
+- [ ] Deployments `README` is updated: old task is moved to the 'deprecated' section, and links are updated with the `deprecated/` prefix
+- [ ] Old task imports in `index.ts`, `input.ts` and fork tests are updated
+- [ ] Old task `README` is updated with a warning sign, a link to the replacement task and a short description <!-- Explain why it was deprecated -->
+
+## Code checklist:
+
+- [ ] The diff is legible and has no extraneous changes <!-- There shouldn't be any changes to contract code in this PR -->
```
Comment (TomAFrench) [inline .github/PULL_REQUEST_TEMPLATE/deployment-preparation-template.md:23]:
```suggestion
- [ ] The diff is legible and has no extraneous changes <!-- There shouldn't be any changes to code outside of the deployment task in this PR -->
```

**Review by TomAFrench — APPROVED** [review]
Not sure that you can choose from multiple templates for PRs (hopefully we can!). This is something we sorely need as deployments are just infrequent enough to forget some of the chores to go with them.

**Comment by EndymionJkb** [comment]
> Not sure that you can choose from multiple templates for PRs (hopefully we can!). This is something we sorely need as deployments are just infrequent enough to forget some of the chores to go with them.

I'm assuming there will be some kind of dropdown - otherwise this feature wouldn't be very useful. What's not clear to me from the docs is whether there is a "default" one (which has to be the same name as the current one).

98% of the PRs are not deployments, so it would be annoying to have to scroll down to choose a "development" one: I'm afraid we would make a lot of mistakes. Also, how does this interact with Draft?

**Comment by TomAFrench** [comment]
@jubeira I'm not seeing a menu to select a template from sadly. We might need to roll this back.

**Pull request merged**